packages/experimental/ActivityIndicator/src/ActivityIndicator.tsx

```diff
@@ -0,0 +1,102 @@
+const useStyling = buildUseStyling(stylingSettings);
+export const ActivityIndicator = compose<ActivityIndicatorType>({
+  displayName: activityIndicatorName,
+  ...stylingSettings,
+  slots: {
+    root: View,
+    svg: AnimatedSvg,
+  },
+  render: (props: ActivityIndicatorProps, useSlots: UseSlots<ActivityIndicatorType>) => {
+    const Slots = useSlots(props);
+    const tokens = useStyling(props).root;
+
+    const memoizedActivityIndicatorData = useMemo(
+      () => ({
+        activityIndicatorColor: props.activityIndicatorColor ? props.activityIndicatorColor : tokens['activityIndicatorColor'],
+        size: props.size ? diameterSizeMap[props.size] : diameterSizeMap[tokens['size']],
+        lineThickness: props.lineThickness
+          ? lineThicknessSizeMap[props.lineThickness]
+          : props.size
+          ? lineThicknessSizeMap[props.size]
+          : lineThicknessSizeMap[tokens['lineThickness']],
+        animating: props.animating != undefined ? props.animating : true,
+        hidesWhenStopped: props.hidesWhenStopped != undefined ? props.hidesWhenStopped : true,
+      }),
+      [props.activityIndicatorColor, props.size, props.lineThickness, props.animating, props.hidesWhenStopped],
+    );
+
+    const spinAnimation = useRef(new Animated.Value(0)).current;
+    useEffect(() => {
+      if (memoizedActivityIndicatorData.animating) {
+        Animated.loop(
+          Animated.timing(spinAnimation, {
+            toValue: 359,
+            duration: 750,
+            useNativeDriver: Platform.OS == 'ios' || Platform.OS == 'android',
+            easing: Easing.linear,
+          }),
+        ).start();
+      }
+    });
+    const interpolateSpin = spinAnimation.interpolate({
+      inputRange: [0, 359],
+      outputRange: ['0deg', '359deg'],
+    });
+
+    // React Native ActivityIndicator still takes up space when hidden, so to perfectly match would use opacity
+    // hiding opacity makes the screen reader on iOS and Android skip over it
+    const hideOpacity = memoizedActivityIndicatorData.animating == false && memoizedActivityIndicatorData.hidesWhenStopped == true ? 0 : 1;
+
+    // Depends on size, line thickness and color after memoization so can't be in the memo itself
+    const path = getActivityIndicatorPath(
+      memoizedActivityIndicatorData.size,
+      memoizedActivityIndicatorData.lineThickness,
+      memoizedActivityIndicatorData.activityIndicatorColor as string,
+    );
+
+    // props for the AnimatedSvg, which is Slots.svg
+    const otherSvgProps = {
+      width: memoizedActivityIndicatorData.size,
+      height: memoizedActivityIndicatorData.size,
+      style: {
+        transform: [{ rotateZ: interpolateSpin }, { perspective: 10 }],
+        opacity: hideOpacity,
+      },
+    };
+
+    const otherRootProps = {
+      style: {
+        opacity: hideOpacity,
+      },
+      accessibilityState: { busy: memoizedActivityIndicatorData.animating },
+    };
```

Comment:
Let's say we removed the call to useMemo altogether, is the logic (minus interpolateSpin) being done here something that could be moved to SlotProps?

Reply:
I ask because 
A) I still think the call to useMemo is mostly unnecessary and we can remove it
B) It feels like what we're doing here is taking the input from `ActivityIndicatorProps` and transforming it into the actual props we want to pass down to each slot, which is what `slotProps` is suppose to be for.

Reply:
Are you talking about the slotProps within the .styling.tsx? I don't think that file takes in any props that the user passes in. These props here all depend on what the user passes or doesn't pass in so I don't think it could be moved to slotProps

Reply:
Ah, that's a good point. Thanks for the clarification!

Reply:
Which reminds me, I think we need to set `tokensThatAreAlsoProps` in the .styling function

Reply:
Do we still have to set that with props extending all of the tokens?

Reply:
Yeah, looking at the readme, it looks like it copies the props that are in the `tokensThatAreAlsoProps` into tokens https://github.com/microsoft/fluentui-react-native/tree/9ae107336b8e6a20dffbfb802facf0f624729fe0/packages/experimental/use-styling

Reply:
Wait, from the readme it seems like it should be `propsThatAreAlsoTokens` because it describes that as "Copy some/none/all props to Tokens" and `tokens = { ...tokens, ...props }`. This confuses me a bit because then, shouldn't I just put all of the "tokens" within props and not have anything within tokens? But I feel like that wouldn't work because then they aren't defined as tokens. I just feel like it's all going in a circle.

Reply:
If I understand correctly, this is because there are multiple places that a token can come from:
- A Theme
- Passed in through `.customize()`
- Passed in as a prop

So you don't necessarily want all your tokens to just be props. You lose the ability for a theme to customize your control that way. That being said, `propsThatAreAlsoTokens` is valid. Fwiw, I think this used to be called `tokenProps` before I complained not knowing what that meant lol.